lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -171,6 +171,12 @@ sub create_from_settings {
         }
     }
 
+    for my $key (keys %settings) {
+        my $value = $settings{$key};
+        $settings{$key} = encode_json($value) if (ref $value eq 'ARRAY' || ref $value eq 'HASH');
+    }
```

Comment:
```suggestion
    %settings = map { my $v = $settings{$_}; ref $v eq 'ARRAY' || ref $v eq 'HASH' ? encode_json($v) : $v } %settings;
```

Reply:
That doesn't work.
You need to iterate over the **keys** of the settings hash, and also you need to return the key **and** the value inside of the map.
We tried your suggestion together and found it not working.
I vote for keeping it like it is.

Reply:
so what is a better way to write code then? Using a loop is fine if we need to abort at some point but here we want an altered copy of the complete hash.

Reply:
I prefer the for loop over your one-liner suggestion as that looks like not the intended use case for map